packages/web/src/components/create-playlist/FolderForm.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { FormEvent, useState } from 'react'
+
+import { Button, IconCheck } from '@audius/stems'
+import { isEmpty } from 'lodash'
+
+import Input from 'components/data-entry/Input'
+
+import styles from './FolderForm.module.css'
+
+const messages = {
+  createFolderButtonText: 'Create Folder'
+}
+
+type FolderFormProps = {
+  onSubmit: (folderName: string) => void
+  initialFolderName?: string
+}
+
+const FolderForm = ({ onSubmit, initialFolderName = '' }: FolderFormProps) => {
+  const [folderName, setFolderName] = useState(initialFolderName)
+  const [inputHasError, setInputHasError] = useState(false)
+  const [hasSubmitted, setHasSubmitted] = useState(false)
+
+  const onChangeFolderName = (newName: string) => {
+    setFolderName(newName)
+  }
+
+  const onFormSubmit = (e: FormEvent) => {
+    e.preventDefault()
+    if (isEmpty(folderName)) {
+      setInputHasError(true)
+    } else {
+      setInputHasError(false)
+      setHasSubmitted(true)
+      onSubmit(folderName)
+    }
+  }
+
+  return (
+    <div className={styles.folderForm}>
+      <form onSubmit={onFormSubmit}>
+        <Input
+          variant='elevatedPlaceholder'
+          placeholder='Folder Name'
+          defaultValue={folderName}
+          error={inputHasError}
+          onChange={onChangeFolderName}
+          characterLimit={64}
+        />
+        <div className={styles.folderFormButtonContainer}>
+          <Button
+            buttonType='submit'
```

Comment:
yess one day indeed

Reply:
i kind of wonder if we should just start all over with a new Stems Button and then gradually migrate all the usage of the old Stems Button to the new one

Reply:
I'm all for that, we just did that with the native button. And since it sounds like there are tons of changes julian requested for the button, might be easiest to start over and slowly adopt

Reply:
SG!